File: ts/768/724.ts
function pivotIndex(nums: number[]): number {
  const len = nums.length
  let ls = 0, rs = 0
  for (let i = 0; i < len; ++i) {
    rs += nums[i]
  }
  for (let i = 0; i < len; ++i) {
    ls += nums[i - 1] || 0
    rs -= nums[i]
    if (ls === rs) return i
  }
  return -1
};

File: c/768/724.c
int pivotIndex(int* nums, int numsSize){
  int total = 0;
  int sum = 0;
  for (int i = 0; i < numsSize; ++i) {
    total += nums[i];
  }
  for (int i = 0; i < numsSize; ++i) {
    if (sum * 2 + nums[i] == total) return i;
    sum += nums[i];
  }
  return -1;
}